DSharpPlus.Commands/Processors/BaseCommandProcessor/BaseCommandProcessor.cs

```diff
@@ -1,184 +1,239 @@
-
-    /// <inheritdoc />
-    public Type ContextType => typeof(TCommandContext);
-
-    /// <inheritdoc />
-    public abstract IReadOnlyList<Command> Commands { get; }
-
-    public virtual void AddConverter<T>(TConverter converter) => AddConverter(typeof(T), converter);
-    public virtual void AddConverter(Type type, TConverter converter) => AddConverter(new() { ParameterType = type, ConverterInstance = converter });
-    public virtual void AddConverters(Assembly assembly) => AddConverters(assembly.GetTypes());
-    public virtual void AddConverters(IEnumerable<Type> types)
+    /// <summary>
+    /// Finds all parameters that are enums and creates a generic enum converter for them.
+    /// </summary>
+    protected virtual void AddEnumConverters()
     {
-        foreach (Type type in types)
+        if (this.extension is null)
         {
-            // Ignore types that don't have a concrete implementation (abstract classes or interfaces)
-            // Additionally ignore types that have open generics (IArgumentConverter<TEventArgs, T>) instead of closed generics (IArgumentConverter<string>)
-            if (type.IsAbstract || type.IsInterface || type.IsGenericTypeDefinition || !type.IsAssignableTo(typeof(TConverter)))
-            {
-                continue;
-            }
-
-            // Check if the type implements IArgumentConverter<TEventArgs, T>
-            Type? genericArgumentConverter = type.GetInterfaces().FirstOrDefault(type => type.IsGenericType && type.GetGenericTypeDefinition() == typeof(IArgumentConverter<,,>));
-            if (genericArgumentConverter is null)
-            {
-                BaseCommandLogging.InvalidArgumentConverterImplementation(this.logger, type.FullName ?? type.Name, typeof(IArgumentConverter<,,>).FullName ?? typeof(IArgumentConverter<,,>).Name, null);
-                continue;
-            }
+            throw new InvalidOperationException("The processor has not been configured yet.");
+        }
 
-            // GenericTypeArguments[2] here is the T in IArgumentConverter<TConverterContext, TEventArgs, T>
-            AddConverter(new() { ParameterType = genericArgumentConverter.GenericTypeArguments[2], ConverterType = type });
+        // Double check to make sure the enum converter can be casted to TConverter.
+        if (!typeof(TConverter).IsAssignableFrom(typeof(EnumConverter<>)))
+        {
+            BaseCommandLogging.invalidEnumConverterImplementation(
+                this.logger,
+                typeof(EnumConverter<>).FullName ?? typeof(EnumConverter<>).Name,
+                typeof(TConverter).FullName ?? typeof(TConverter).Name,
+                GetType().FullName ?? GetType().Name,
+                nameof(AddEnumConverters),
+                null
+            );

```

Comment:
this doesn't seem to actually make any attempt at ensuring the enum converter can be casted, it just logs and continues